x/wasm/types/interface_registry.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	"github.com/cosmos/gogoproto/proto"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+)
+
+var _ codectypes.InterfaceRegistry = &WasmInterfaceRegistry{}
+
+// NewWasmInterfaceRegistry returns a new WasmInterfaceRegistry instance
+func NewWasmInterfaceRegistry(registry codectypes.InterfaceRegistry) WasmInterfaceRegistry {
+	return WasmInterfaceRegistry{registry}
+}
+
+// WasmInterfaceRegistry represents an interface registry with a custom any resolver
+type WasmInterfaceRegistry struct {
+	codectypes.InterfaceRegistry
+}
+
+// Resolve implements codectypes.InterfaceRegistry
+func (WasmInterfaceRegistry) Resolve(typeUrl string) (proto.Message, error) {
+	return new(WasmAny), nil
+}
```

Comment:
Doesn't this make this registry pretty much do nothing?

Reply:
Hmm, I thought this makes it have a custom handler for the `WasmAny` type (force a custom `Any` type) and all others handled normally

Reply:
This would make all message types resolve to a `WasmAny`, regardless of what's asked for.
If you're trying to make it use a custom any instead of normal any, you'd have to check for that specific `typeURL`, then either return this new any, or else use the `<receiver>.InterfaceRegistry.Resolve(typeUrl)`.

Reply:
Looks like others are doing it this way too. So it must be okay to resolve everything to that new `WasmAny`.